test/e2e/storage/csi_volumes.go

```diff
@@ -245,16 +249,156 @@ var _ = utils.SIGDescribe("CSI Volumes", func() {
+				driverExists:  false,
+				expectPodInfo: false,
+			},
+		}
+
+		for _, t := range tests {
+			test := t
+			It(test.name, func() {
+				if test.driverExists {
+					driver := createCSIDriver(csics, "csi-mock-"+f.UniqueName, true, test.podInfoOnMountVersion)
+					if driver != nil {
+						defer csics.CsiV1alpha1().CSIDrivers().Delete(driver.Name, nil)
+					}
+				}
+
+				By("Creating pod")
+				t := driver.createStorageClassTest()
+				class, claim, pod := startPausePod(cs, t, ns.Name)
+				if class != nil {
+					defer cs.StorageV1().StorageClasses().Delete(class.Name, nil)
+				}
+				if claim != nil {
+					defer cs.CoreV1().PersistentVolumeClaims(ns.Name).Delete(claim.Name, nil)
+				}
+				if pod != nil {
+					// Fully delete (=unmount) the pod before deleting CSI driver
+					defer framework.DeletePodWithWait(f, cs, pod)
+				}
+				if pod == nil {
+					return
+				}
+
+				err := framework.WaitForPodNameRunningInNamespace(cs, pod.Name, pod.Namespace)
+				framework.ExpectNoError(err, "Failed to start pod: %v", err)
+
+				By("Checking CSI driver logs")
+				// The driver is deployed as a statefulset with stable pod names
+				driverPodName := "csi-mockplugin-0"
+				log, err := framework.GetPodLogs(cs, f.Namespace.Name, driverPodName, "mock")
+				framework.ExpectNoError(err, "Could not load CSI driver logs")
+				framework.Logf("CSI driver logs:\n%s", log)
+
+				expectedAttributes := map[string]string{
+					"csi.storage.k8s.io/pod.name":            pod.Name,
+					"csi.storage.k8s.io/pod.namespace":       pod.Namespace,
+					"csi.storage.k8s.io/pod.uid":             string(pod.UID),
+					"csi.storage.k8s.io/serviceAccount.name": "default",
+				}
+				foundAttributes := sets.NewString()
+				logLines := strings.Split(log, "\n")
+				for _, line := range logLines {
+					if !strings.HasPrefix(line, "gRPCCall:") {
+						continue
+					}
+					line = strings.TrimPrefix(line, "gRPCCall:")
+					// Dummy structure that parses just volume_attributes out of logged CSI call
+					type MockCSICall struct {
+						Method  string
+						Request struct {
+							VolumeAttributes map[string]string `json:"volume_attributes"`
+						}
+					}
+					var call MockCSICall
+					err := json.Unmarshal([]byte(line), &call)
+					if err != nil {
+						framework.ExpectNoError(err, "Could not parse CSI driver log line", line)
+						continue
+					}
+
+					if call.Method != "/csi.v0.Node/NodePublishVolume" {
+						continue
+					}
+					for k, v := range expectedAttributes {
+						vv, found := call.Request.VolumeAttributes[k]
+						if found && v == vv {
+							foundAttributes.Insert(k)
+							framework.Logf("Found volume attribute %s: %s", k, v)
+						}
+					}
+					// Process just the first NodePublish, the rest of the log is useless.
+					break
+				}
+				if test.expectPodInfo {
+					Expect(foundAttributes.Len()).To(Equal(len(expectedAttributes)), "Number of found volume attributes does not match")
```

Comment:
@msau42, I get an error here because it doesn't find any of the `expectedAttributes`. The only volume attribute that I get is `storage.kubernetes.io/csiProvisionerIdentity`. This happens in both GCE and locally:

```
{
    "Method": "/csi.v0.Node/NodePublishVolume",
...
        "volume_attributes": {
            "name": "pvc-d7d839c6-e1d3-11e8-9654-42010af00019",
            "storage.kubernetes.io/csiProvisionerIdentity": "1541516063008-8081-csi-mock-e2e-tests-csi-volumes-5mjl2"
        }
    }
...
}
```

Also, I checked a volumed created the EBS driver and this seems to be the only attribute added as well: https://gist.github.com/bertinatto/3a42384ddcc9c60dcc2ce650a975afce#file-pv_csi_ebs-yaml-L34

Does this look like a bug to you as well? Or am I missing something?

Reply:
Update: it seems like for some reason k8s is not injecting the pod information into volume_attributes: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_mounter.go#L257

Reply:
Update: oh, apparently I need to enable the feature gate: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_mounter.go#L235

Reply:
Update: yes, the problem was the gate. Sorry for the noise.